src/Analyzer.Cli/Loggers/SummaryLogger.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.Templates.Analyzer.Cli
+{
+    /// <summary>
+    /// Class to print a summary of the execution logs
+    /// </summary>
+    public class SummaryLogger : ILogger
+    {
+        private readonly Dictionary<string, int> loggedErrors = new();
+        private readonly Dictionary<string, int> loggedWarnings = new();
+
+        /// <summary>
+        /// Constructor of the SummaryLogger class
+        /// </summary>
+        public SummaryLogger()
+        {
+        }
+
+        /// <inheritdoc/>
+        public IDisposable BeginScope<TState>(TState state) => default!;
+
+        /// <inheritdoc/>
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return true;
+        }
+
+        /// <inheritdoc/>
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            var recordLog = new Action<Dictionary<string, int>, string>((logs, newLogMessage) => {
+                if (!logs.TryAdd(newLogMessage, 1))
+                {
+                    logs[newLogMessage] += 1;
+                }
+            });
+
+            var logMessage = state.ToString();
+
+            if (logLevel == LogLevel.Error)
+            {
+                recordLog(loggedErrors, logMessage);
+            }
+            else if (logLevel == LogLevel.Warning)
+            {
+                recordLog(loggedWarnings, logMessage);
+            }
+        }
+
+        /// <summary>
+        /// Outputs to console a summary of the errors and warnings logged
+        /// </summary>
+        public void SummarizeLogs(bool showVerboseMessage)
+        {
+            if (loggedErrors.Count > 0 || loggedWarnings.Count > 0)
+            {
+                Console.ForegroundColor = loggedErrors.Count > 0 ? ConsoleColor.Red : ConsoleColor.Yellow;
+
+                Console.WriteLine($"{Environment.NewLine}{loggedErrors.Count} error(s) and {loggedWarnings.Count} warning(s) were found during the execution, please refer to the original messages above");
+
+                if (!showVerboseMessage)
+                {
+                    Console.WriteLine("The verbose mode (option -v or --verbose) can be used to obtain even more information about the execution");
```

Comment:
So this would be red or yellow... Does that make sense?

Reply:
In the current state, it depends on whether there are errors present - if there were errors, it would be red; otherwise yellow.